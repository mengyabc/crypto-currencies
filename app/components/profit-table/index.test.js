import React from 'react'
import { shallow } from 'enzyme'

import ProfitTable from '.'

describe('BestProfit component', () => {
  test('Should render correctly', () => {
    const result = {
      currency: 'BTC',
      date: '20180507',
      buy: {
        time: '0915',
        price: '34.98',
      },
      sell: {
        time: '1230',
        price: '37.01',
      },
      profit: '2.03',
    }
    const wrapper = shallow(<ProfitTable result={result} />)
    expect(wrapper).toMatchSnapshot()
  })
})
