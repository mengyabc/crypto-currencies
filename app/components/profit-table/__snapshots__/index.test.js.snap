// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestProfit component Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfitTable
    result={
      Object {
        "buy": Object {
          "price": "34.98",
          "time": "0915",
        },
        "currency": "BTC",
        "date": "20180507",
        "profit": "2.03",
        "sell": Object {
          "price": "37.01",
          "time": "1230",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table
        className="table table-bordered text-center"
      >
        <thead>
          <tr>
            <td
              colSpan="2"
            >
              07-May-18
            </td>
          </tr>
          <tr>
            <th
              className="table-bg"
              colSpan="2"
            >
              BTC
            </th>
          </tr>
          <tr>
            <th>
              buy
            </th>
            <th>
              sell
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              $34.98
            </td>
            <td>
              $37.01
            </td>
          </tr>
          <tr>
            <td>
              09:15 AM
            </td>
            <td>
              12:30 PM
            </td>
          </tr>
          <tr>
            <td
              className="text-center"
              colSpan="2"
            >
              Profit: $2.03
            </td>
          </tr>
        </tbody>
      </table>,
      "className": "col-xs-12 col-md-4",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <td
                colSpan="2"
              >
                07-May-18
              </td>
            </tr>
            <tr>
              <th
                className="table-bg"
                colSpan="2"
              >
                BTC
              </th>
            </tr>
            <tr>
              <th>
                buy
              </th>
              <th>
                sell
              </th>
            </tr>
          </thead>,
          <tbody>
            <tr>
              <td>
                $34.98
              </td>
              <td>
                $37.01
              </td>
            </tr>
            <tr>
              <td>
                09:15 AM
              </td>
              <td>
                12:30 PM
              </td>
            </tr>
            <tr>
              <td
                className="text-center"
                colSpan="2"
              >
                Profit: $2.03
              </td>
            </tr>
          </tbody>,
        ],
        "className": "table table-bordered text-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <td
                  colSpan="2"
                >
                  07-May-18
                </td>
              </tr>,
              <tr>
                <th
                  className="table-bg"
                  colSpan="2"
                >
                  BTC
                </th>
              </tr>,
              <tr>
                <th>
                  buy
                </th>
                <th>
                  sell
                </th>
              </tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <td
                  colSpan="2"
                >
                  07-May-18
                </td>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "07-May-18",
                  "colSpan": "2",
                },
                "ref": null,
                "rendered": "07-May-18",
                "type": "td",
              },
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <th
                  className="table-bg"
                  colSpan="2"
                >
                  BTC
                </th>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "BTC",
                  "className": "table-bg",
                  "colSpan": "2",
                },
                "ref": null,
                "rendered": "BTC",
                "type": "th",
              },
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    buy
                  </th>,
                  <th>
                    sell
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "buy",
                  },
                  "ref": null,
                  "rendered": "buy",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "sell",
                  },
                  "ref": null,
                  "rendered": "sell",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <td>
                  $34.98
                </td>
                <td>
                  $37.01
                </td>
              </tr>,
              <tr>
                <td>
                  09:15 AM
                </td>
                <td>
                  12:30 PM
                </td>
              </tr>,
              <tr>
                <td
                  className="text-center"
                  colSpan="2"
                >
                  Profit: $2.03
                </td>
              </tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    $34.98
                  </td>,
                  <td>
                    $37.01
                  </td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "$34.98",
                  },
                  "ref": null,
                  "rendered": "$34.98",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "$37.01",
                  },
                  "ref": null,
                  "rendered": "$37.01",
                  "type": "td",
                },
              ],
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    09:15 AM
                  </td>,
                  <td>
                    12:30 PM
                  </td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "09:15 AM",
                  },
                  "ref": null,
                  "rendered": "09:15 AM",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "12:30 PM",
                  },
                  "ref": null,
                  "rendered": "12:30 PM",
                  "type": "td",
                },
              ],
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <td
                  className="text-center"
                  colSpan="2"
                >
                  Profit: $2.03
                </td>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Profit: $2.03",
                  "className": "text-center",
                  "colSpan": "2",
                },
                "ref": null,
                "rendered": "Profit: $2.03",
                "type": "td",
              },
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table
          className="table table-bordered text-center"
        >
          <thead>
            <tr>
              <td
                colSpan="2"
              >
                07-May-18
              </td>
            </tr>
            <tr>
              <th
                className="table-bg"
                colSpan="2"
              >
                BTC
              </th>
            </tr>
            <tr>
              <th>
                buy
              </th>
              <th>
                sell
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                $34.98
              </td>
              <td>
                $37.01
              </td>
            </tr>
            <tr>
              <td>
                09:15 AM
              </td>
              <td>
                12:30 PM
              </td>
            </tr>
            <tr>
              <td
                className="text-center"
                colSpan="2"
              >
                Profit: $2.03
              </td>
            </tr>
          </tbody>
        </table>,
        "className": "col-xs-12 col-md-4",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <td
                  colSpan="2"
                >
                  07-May-18
                </td>
              </tr>
              <tr>
                <th
                  className="table-bg"
                  colSpan="2"
                >
                  BTC
                </th>
              </tr>
              <tr>
                <th>
                  buy
                </th>
                <th>
                  sell
                </th>
              </tr>
            </thead>,
            <tbody>
              <tr>
                <td>
                  $34.98
                </td>
                <td>
                  $37.01
                </td>
              </tr>
              <tr>
                <td>
                  09:15 AM
                </td>
                <td>
                  12:30 PM
                </td>
              </tr>
              <tr>
                <td
                  className="text-center"
                  colSpan="2"
                >
                  Profit: $2.03
                </td>
              </tr>
            </tbody>,
          ],
          "className": "table table-bordered text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr>
                  <td
                    colSpan="2"
                  >
                    07-May-18
                  </td>
                </tr>,
                <tr>
                  <th
                    className="table-bg"
                    colSpan="2"
                  >
                    BTC
                  </th>
                </tr>,
                <tr>
                  <th>
                    buy
                  </th>
                  <th>
                    sell
                  </th>
                </tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <td
                    colSpan="2"
                  >
                    07-May-18
                  </td>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "07-May-18",
                    "colSpan": "2",
                  },
                  "ref": null,
                  "rendered": "07-May-18",
                  "type": "td",
                },
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <th
                    className="table-bg"
                    colSpan="2"
                  >
                    BTC
                  </th>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "BTC",
                    "className": "table-bg",
                    "colSpan": "2",
                  },
                  "ref": null,
                  "rendered": "BTC",
                  "type": "th",
                },
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      buy
                    </th>,
                    <th>
                      sell
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "buy",
                    },
                    "ref": null,
                    "rendered": "buy",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "sell",
                    },
                    "ref": null,
                    "rendered": "sell",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr>
                  <td>
                    $34.98
                  </td>
                  <td>
                    $37.01
                  </td>
                </tr>,
                <tr>
                  <td>
                    09:15 AM
                  </td>
                  <td>
                    12:30 PM
                  </td>
                </tr>,
                <tr>
                  <td
                    className="text-center"
                    colSpan="2"
                  >
                    Profit: $2.03
                  </td>
                </tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                      $34.98
                    </td>,
                    <td>
                      $37.01
                    </td>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "$34.98",
                    },
                    "ref": null,
                    "rendered": "$34.98",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "$37.01",
                    },
                    "ref": null,
                    "rendered": "$37.01",
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                      09:15 AM
                    </td>,
                    <td>
                      12:30 PM
                    </td>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "09:15 AM",
                    },
                    "ref": null,
                    "rendered": "09:15 AM",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "12:30 PM",
                    },
                    "ref": null,
                    "rendered": "12:30 PM",
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <td
                    className="text-center"
                    colSpan="2"
                  >
                    Profit: $2.03
                  </td>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Profit: $2.03",
                    "className": "text-center",
                    "colSpan": "2",
                  },
                  "ref": null,
                  "rendered": "Profit: $2.03",
                  "type": "td",
                },
                "type": "tr",
              },
            ],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
